refactor(pallet-gear-builtin) builtin ids can now be read by subscan

Lazark0x opened the pull request:
Resolves #4694

Lazark0x committed:
initial

Lazark0x committed:
Merge remote-tracking branch 'origin/master' into laz/4694-builtin-ids-as-pallet-ids

# Conflicts:
#	examples/proxy-broker/src/wasm.rs
#	pallets/gear-builtin/src/lib.rs
#	pallets/gear-builtin/src/mock.rs
#	pallets/gear-builtin/src/tests/bad_builtin_ids.rs
#	pallets/gear-builtin/src/tests/staking.rs
#	runtime/vara/Cargo.toml
#	runtime/vara/src/lib.rs

Lazark0x committed:
reformat

Lazark0x committed:
better builtin address formatting

Lazark0x committed:
Merge branch 'master' into laz/4694-builtin-ids-as-pallet-ids

Lazark0x committed:
removed dependencies

Lazark0x committed:
Merge remote-tracking branch 'origin/master' into laz/4694-builtin-ids-as-pallet-ids

Lazark0x committed:
qf

Lazark0x committed:
Merge remote-tracking branch 'origin/master' into laz/4694-builtin-ids-as-pallet-ids

# Conflicts:
#	pallets/gear-builtin/src/mock.rs
#	pallets/gear-builtin/src/tests/bad_builtin_ids.rs
#	pallets/gear-eth-bridge/src/builtin.rs

Lazark0x committed:
Merge remote-tracking branch 'origin/master' into laz/4694-builtin-ids-as-pallet-ids

# Conflicts:
#	pallets/gear-builtin/src/tests/basic.rs
#	pallets/gear-builtin/src/tests/staking.rs

techraed commented on pallets/gear-builtin/src/lib.rs:
```diff
@@ -85,6 +87,148 @@ pub type GasAllowanceOf<T> = <<T as Config>::BlockLimiter as BlockLimiter>::GasA
 
+#[derive(Clone, Copy, Default, Eq, PartialEq, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+pub struct BuiltinActorId {
```
what about moving this stuff along with BuiltinActorType to the gbuiltins?
Ids are demanded by sails

Lazark0x committed:
Merge branch 'master' into laz/4694-builtin-ids-as-pallet-ids

# Conflicts:
#	pallets/gear-builtin/Cargo.toml

Lazark0x committed:
Merge branch 'master' into laz/4694-builtin-ids-as-pallet-ids

Lazark0x committed:
returned serde

Lazark0x committed:
Merge master into laz/4694-builtin-ids-as-pallet-ids

Resolved conflicts:
- pallets/gear-builtin/src/tests/bad_builtin_ids.rs: kept assert_ok import, removed ConstBool
- pallets/gear-eth-bridge/src/builtin.rs: added QueueId import and kept module doc comment
- runtime/vara/Cargo.toml: kept dev feature with pallet-gear-eth-bridge
- runtime/vara/src/lib.rs: kept conditional imports and added eth-bridge to BuiltinActors

Lazark0x committed:
Fix dev feature in vara-runtime Cargo.toml

Remove pallet-gear-eth-bridge from dev feature list as it's not an optional dependency

Lazark0x committed:
Remove incorrect cfg attribute from BuiltinActors type

The BuiltinActors type must be unconditionally available for WASM runtime builds.
Removed #[cfg(not(feature = "dev"))] that was incorrectly added during merge conflict resolution.

Lazark0x committed:
Merge remote-tracking branch 'origin/master' into laz/4694-builtin-ids-as-pallet-ids

Lazark0x committed:
moved BuiltinActorId into separate crate inside gbuiltins common, refactoring and improvements

Lazark0x committed:
fixes

Lazark0x committed:
lock updated

Lazark0x committed:
Merge remote-tracking branch 'origin/master' into laz/4694-builtin-ids-as-pallet-ids

# Conflicts:
#	Cargo.lock
#	examples/bls381/Cargo.toml
#	pallets/gear-builtin/Cargo.toml
#	pallets/gear-builtin/rpc/runtime-api/Cargo.toml
#	pallets/gear-builtin/src/lib.rs
#	pallets/gear-builtin/src/mock.rs
#	pallets/gear-builtin/src/tests/bad_builtin_ids.rs
#	pallets/gear-eth-bridge/Cargo.toml
#	pallets/gear-eth-bridge/src/builtin.rs

Lazark0x committed:
little refactoring

breathx commented:
Closing for now as it's not a current priority — can be revisited later.
